modules/gridmap/grid_map_editor_plugin.cpp

```diff
@@ -54,8 +56,18 @@ void GridMapEditor::_configure() {
 
 void GridMapEditor::_menu_option(int p_option) {
 
+	bool using_freelook = Input::get_singleton()->is_mouse_button_pressed(BUTTON_RIGHT);
```

Comment:
Isn't there a way to know that without hardcoding an input check? @Zylann is about to add support for toggling freelook with a hotkey.

Reply:
I agree that you shouldn't imply RMB means freelook

Reply:
yes thats why i added the flag, its true, that freelook mode could be toggled in different ways.

Reply:
That means we would need to change Gridmap if freelook changes... which isn't good design (not DRY and no SoC). Currently the only true way to access this info is with `SpatialEditorViewport::is_freelook_active()` though...

Reply:
@Zylann that flag is only used to avoid rotation of tiles, if frelook, or any other shortcut / mode, changes grid map and other stuff will need to change (that flag might even be irrelevant on such cases), because of the possible conflicts, so this is not about design, godot works this way, maybe later we could have a better input handling mechanism.

Reply:
Note that when I say "consume events", Godot already supports this. If using that feature isn't enough then I don't know... but I can say your check will fail as soon as I PR the feature proposed in https://github.com/godotengine/godot/issues/10463#issuecomment-340283571

Reply:
@Zylann also consider that such navigation mode, will make grid map editing almost unusable, or very hard at least (it will also override QWER keys for spatial gizmos), but yeah, im not sure how people will use it, also im not sure how to fix that in a smarter way, i mean there will be something here that requires knowledge of the general viewport shortcuts.